main/src/ca/uwaterloo/flix/language/phase/MonoDefs.scala

```diff
@@ -273,19 +272,42 @@ object MonoDefs {
       }
     }
 
+    val effects = ParOps.parMapValues(root.effects){
+      case LoweredAst.Effect(doc, ann, mod, sym, ops0, loc) =>
+        val ops = ops0.map{
+          case LoweredAst.Op(sym, spec) =>
+            MonoAst.Op(sym, visitEffectOpSpec(spec, empty))
+        }
+        MonoAst.Effect(doc, ann, mod, sym, ops, loc)
+    }
+
     // Reassemble the AST.
-    root.copy(
-      defs = ctx.toMap,
-      classes = Map.empty,
-      instances = Map.empty,
-      sigs = Map.empty
+    MonoAst.Root(
+      ctx.toMap,
+      effects,
+      root.entryPoint,
+      root.reachable, // does this make sense?
+      root.sources
     )
   }
 
+  /**
+    * converts the given effect op spec. Assumes that the spec has no variables,
```

Comment:
I don't like the word "assumes".

Is this correct or not? @mlutze ?

Reply:
Ops are indeed monomorphic atm so this is a sound assumption.